py-rse/ci.Rmd

```diff
@@ -4,42 +4,39 @@
 source(here::here("_common.R"))
 ```
 
-[Continuous integration][continuous-integration] (CI) a simple idea:
-merge and test changes to a project automatically and frequently
-so that developers get feedback about problems
-while those changes are still fresh in their minds.
+[Continuous integration][continuous-integration] (CI) is a relatively simple idea:
+run checks and tests automatically whenever changes have been made to a project.
+This then gives immediate feedback to the project developers if the changes cause potential problems, 
+that way its easier to fix the problem by seeing what was changed.
 Automation tools like Make (Chapter \@ref(py-rse-automate)) can compile software,
-run tests,
-re-do analyses,
+run tests, re-do analyses,
 and re-create reports with a single command;
-CI tools run those commands and give users feedback every time
+CI tools can run those commands and give users feedback every time
 something changes in the project's repository.
 These tools can also be set up to run tests with several different configurations of the software
 or on several different operating systems,
-so that (for example) if a programmer makes a change on Windows
-that would break something for Mac users or vice versa,
-they find out before their friends' lives are disrupted.
+so that if (for example) a programmer makes a change on Windows 
+and it then breaks only for Mac users or vice versa,
+they can find out before releasing the software to others.
 
 This chapter introduces a CI tool called [Travis CI][travis-ci]
 that integrates well with [GitHub][github].
 If Travis CI has been set up,
 then every time a change is committed to a GitHub repository,
 Travis CI creates a fresh environment,
-checks out a copy of the project,
+makes a copy of the project (by cloning),
```

Comment:
"make a fresh clone of the repository" [with crossref to our discussion of cloning]